src/System.AspNetCore/TimedScopesExtensions.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Omex.System.Diagnostics;
+using Microsoft.Omex.System.TimedScopes;
+using Microsoft.Omex.System.TimedScopes.ReplayEventLogging;
+
+namespace Microsoft.Omex.System.AspNetCore
+{
+	/// <summary>
+	/// Extension methods for the TimedScopes
+	/// </summary>
+	public static class TimedScopesExtensions
+	{
+		/// <summary>
+		/// Register types required for timedscopes
+		/// </summary>
+		public static IServiceCollection AddTimedScopes(this IServiceCollection serviceCollection)
+		{
+			return serviceCollection
+				.AddSingleton<IMachineInformation, BasicMachineInformation>()
```

Comment:
>IMachineInformation [](start = 18, length = 19)

Is there a logic to the ordering? If not, I'd advise sorting alphabetically as there it avoids someone accidentally adding the same dependency twice, which can happen surprisingly often - particularly if two people are modifying the code at the same time